pkg/api/server/cel2sql/interpreter.go

```diff
@@ -293,7 +293,6 @@ func (i *interpreter) interpretUnaryCallExpr(expr *exprpb.Expr_Call) error {
 	if err := i.interpretExpr(expr.Args[0]); err != nil {
 		return err
 	}
-	i.query.WriteString(space)
```

Comment:
I'm somewhat curious - why don't we need an extra space at the end?

It may be worth a long-form comment in the code (or godoc) which provides an example input/output of this function.

Reply:
This was not caught yet because we didn't utilize the unary call at all.
 
https://github.com/tektoncd/results/blob/95d3eb8b5afa150b45c0e73306a142046604574b/pkg/api/server/cel2sql/interpreter.go#L289-L298

In the current code on line 296 we are adding an extra space because we have already added a space on line 292. For normal queries, it doesn't change anything. But if we use the NOT operator in complex queries, this happens

```
=== RUN   TestConvertRecordExpressions/complex_not_expressions
    convert_test.go:160: want: NOT POSITION('foo' IN (data->'metadata'->>'annotations')) <> 0 AND (data->'metadata'->>'name') LIKE '%' || 'bar'
    convert_test.go:161: got:  NOT POSITION('foo' IN (data->'metadata'->>'annotations')) <> 0  AND (data->'metadata'->>'name') LIKE '%' || 'bar'
    convert_test.go:163: Mismatch (-want +got):
          strings.Join({
          	"NOT POSITION('foo' IN (data->'metadata'->>'annotations')) <> 0",
        + 	" ",
          	" AND (data->'metadata'->>'name') LIKE '%' || 'bar'",
          }, "")
    --- FAIL: TestConvertRecordExpressions/complex_not_expressions (0.00s)
```
the resulting SQL query has **two** spaces between 0 and AND. Also if you see other functions (binary interpreter for reference), the space is not added after the query is interpreted.